api/rest/cluster_bundle_handler.go

```diff
@@ -31,7 +31,26 @@ type ClusterBundleHandler struct {
 }
 
 func NewClusterBundleHandler(c Coordinator, client Client, tools dcos.Tooler, workDir string, timeout time.Duration,
-	urlBuilder dcos.NodeURLBuilder) *ClusterBundleHandler {
+	urlBuilder dcos.NodeURLBuilder) (*ClusterBundleHandler, error) {
+	f, err := os.Stat(workDir)
```

Comment:
Since this is the exact same code as above in `NewBundleHandler` you should outsource this into a func, e.g. `ensureWorkDirExists()`.